packages/forms/src/model/form_group.ts

```diff
@@ -632,7 +632,14 @@ export const isFormGroup = (control: unknown): control is FormGroup => control i
  * @publicApi
  */
 export class FormRecord<TControl extends AbstractControl = AbstractControl> extends
-    FormGroup<{[key: string]: TControl}> {}
+    FormGroup<{[key: string]: TControl}> {
+  /**
+   * Internal marker to differentiate `FormRecord` from `FormGroup` type.
+   *
+   * This is for internal Angular usage to support typed forms; do not directly use it.
+   */
+  private readonly ɵformRecordBrand = Symbol('FormRecordBrand');
```

Comment:
This probably doesn't need a "real" value assignment; only having it on the type-side should be sufficient:

```suggestion
  private readonly brand!: symbol;
```

or even

```suggestion
  private readonly brand!: unknown;
```

Since the property is private the latin-barred-o prefix is also not really needed.

Reply:
You're right!

It's done (I kept the name `formRecordBrand`, because the goal is also to differentiate from potential other subclasses...)